### PR TITLE
#175 - Optimized, common selection algorithm for all navigation methods

### PR DESCRIPTION
**Pull request by tfesenko**
swagger_selection

### CODE REVIEW

**Comment by tfesenko** [comment]
@andylowry , can you please review?

File: com.reprezen.swagedit/src/com/reprezen/swagedit/model/AbstractNode.java [hunk 1]
```diff
@@ -175,7 +176,11 @@ public Position getPosition(IDocument document) {
         int endCol = getEnd().getColumn();
         try {
             offset = document.getLineOffset(startLine);
-            length = (document.getLineOffset(endLine) + endCol) - offset;
+            if (selectEntireElement) {
+                length = (document.getLineOffset(endLine) + endCol) - offset;
+            } else {
+                length = document.getLineOffset(startLine+1) - offset;
```
Comment (andylowry) [inline com.reprezen.swagedit/src/com/reprezen/swagedit/model/AbstractNode.java:182]: Coulld this throw BadLocationException if startLine is the last line in the document?

**Review by andylowry — COMMENTED** [review]
@tfesenko One question for you to consider and handle as needed. Otherwise looks good.

**Comment by andylowry** [comment]
@tfesenko I've fixed the edge case I pointed out. It turns out it's very unlikely. You need a single-line object and no line terminator. Easiest way to test is with an empty method, e.g.:

```
  path /foo:
    get: {}
````
If there's no following line-terminator, then the old code hits BadLocationException and flips back to the top of the editor. With the fix you get the get method highlighted as expected. Obviously not a very important edge case in retrospect, but it's fixed, so please review / test ASAP so we can get a release out for Finnova.

Thanks
